vtkext/private/module/vtkF3DEGLRenderWindow.cxx

```diff
@@ -0,0 +1,29 @@
+#include <vtkObjectFactory.h>
+#include <vtkVersion.h>
+
+#if VTK_VERSION_NUMBER >= VTK_VERSION_CHECK(9, 3, 20240914)
+#include <vtk_glad.h>
+#include <vtkglad/include/glad/egl.h>
+#endif
+
+#include "vtkF3DEGLRenderWindow.h"
+
+//------------------------------------------------------------------------------
+vtkF3DEGLRenderWindow::vtkF3DEGLRenderWindow() = default;
+
+//------------------------------------------------------------------------------
+vtkF3DEGLRenderWindow::~vtkF3DEGLRenderWindow() = default;
+
+//------------------------------------------------------------------------------
+vtkF3DEGLRenderWindow* vtkF3DEGLRenderWindow::New()
+{
+#if VTK_VERSION_NUMBER >= VTK_VERSION_CHECK(9, 3, 20240914)
+  // Load core egl functions
+  if (gladLoaderLoadEGL(EGL_NO_DISPLAY) == 0)
+  {
+    return nullptr;
```

Comment:
contribute that to VTK ?

Reply:
There's already such logic in `vtkOpenGLRenderWindow::New()` function.  
I had to create our own complete logic to avoid having to raise a warning when X is not detected.  
I decided to subclass the EGL render window to add the `glad` stuff in there to mimic the GLX class.  
Also it would be convenient in case we decide to change the behavior of the VTK EGL window at some point.

Reply:
Ill discuss with Jaswant to see what he thinkgs there

Reply:
Right, please take over on this. I'm sure the logic would be different in VTK. Closing the discussion